src/sentry_json.c

```diff
@@ -220,8 +221,16 @@ sentry__jsonwriter_write_double(sentry_jsonwriter_t *jw, double val)
     if (can_write_item(jw)) {
         char buf[50];
         // The MAX_SAFE_INTEGER is 9007199254740991, which has 16 digits
-        sentry__snprintf_c(buf, sizeof(buf), "%.16g", val);
-        write_str(jw, buf);
+        size_t written
+            = (size_t)sentry__snprintf_c(buf, sizeof(buf), "%.16g", val);
+        // print `null` if we have printf issues or a non-finite double, which
+        // can't be represented in JSON.
+        if (written < 0 || written >= sizeof(buf) || !isfinite(val)) {
+            write_str(jw, "null");
+        } else {
+            buf[written] = '\0';
```

Comment:
Why is this necessary?

Reply:
just to be safe. some invalid session payloads look like strings were not terminated correctly, and I remember reading somewhere that there are bugs around null-termination, also this here: https://stackoverflow.com/a/13067917